tests/test_distribution_ops.py

```diff
@@ -36,3 +38,50 @@ def test_accuracy_exponential_(shape, dtype):
     with flag_gems.use_gems():
         x.exponential_()
     assert x.min() > 0
+
+
+@pytest.mark.parametrize("shape", [(1000,), (100, 1000)])
+@pytest.mark.parametrize("dtype", FLOAT_DTYPES)
+@pytest.mark.parametrize("n_samples", [2048])
+def test_accuracy_multinomial_with_replacement(shape, dtype, n_samples):
+    dist = torch.zeros(size=shape, dtype=dtype, device="cuda")
+    Index = [5, 13, 42]
+    dist[..., Index] = 1
+    with flag_gems.use_gems():
+        res_out = torch.multinomial(dist, n_samples, True)
+    index, tally = torch.unique(res_out, sorted=True, return_counts=True)
+    assert index.tolist() == Index
+    # Do a simple Chi-square test
+    tally = np.array(tally.tolist())
+    expected = tally
+    expected[:] = tally.mean()
```

Comment:
why is the mean number expected?

Reply:
Here I devised a test where the three indices are sampled uniformly with replacement and then test the uniformity of the output with a Chi-square test.  This is equivalent to testing samples of a categorical distribution. To use Chi-square. we need to supply the actual sample counts and the expected sample counts. Since the original probabilities are evenly divided between the three indices the expected sample count for each category is just the mean.

Reply:
As to why using Chi-square to test accuracy, the short answer is Chi-square is quite often used verifying if a large number of samples are drawn from a categorical distribution. The multinomial function in Pytorch is no other than sampling from a given categorical distribution.

Reply:
got it